rebasehelper/tags.py

```diff
@@ -82,23 +84,49 @@ def _map_sections_to_parsed(cls, raw_content: SpecContent, parsed_content: SpecC
 
     @classmethod
     def _parse(cls, raw_content: SpecContent, parsed_content: SpecContent) -> Tuple[Tag]:
-        result = []
+        result: List[Tag] = []
         parsed_mapping = cls._map_sections_to_parsed(raw_content, parsed_content)
+        patch_index = 0
+        source_index = 0
         # counts number of occurrences of one particular section
         for section_index, (section, section_content) in enumerate(raw_content.sections):
             section = section.lower()
             parsed_section_index = parsed_mapping[section_index]
             parsed: List[str] = [] if parsed_section_index == -1 else parsed_content.sections[parsed_section_index][1]
             if section.startswith('%package'):
-                result.extend(cls._parse_package_tags(section, section_content, parsed, section_index))
+                new_tags, patch_index, source_index = cls._parse_package_tags(section, section_content, parsed,
+                                                                              section_index, patch_index, source_index)
+                result.extend(new_tags)
             elif section.startswith('%sourcelist') or section.startswith('%patchlist'):
-                result.extend(cls._parse_list_tags(section, section_content, parsed,
-                                                   cast(Tuple[Tag], tuple(result)), section_index))
+                new_tags, patch_index, source_index = cls._parse_list_tags(section, section_content, parsed,
+                                                                           section_index, patch_index, source_index)
+                result.extend(new_tags)
         return cast(Tuple[Tag], tuple(result))
 
+    @classmethod
+    def _sanitize_tag(cls, name: str, patch_index: int, source_index: int) -> Tuple[str, int, int, Optional[int]]:
+        """Sanitizes a tag name.
+
+        Capitalizes the tag. If the tag is numbered, strips padded zeroes.
+        Implements automatic numbering of Patches and Sources (adds the number
+        to their name).
+        """
+        name = name.lower()
+        index = None
+        tokens = re.split(r'(\d+)', name, 1)
+        if name.startswith('source'):
+            index = source_index if len(tokens) == 1 else int(tokens[1])
+            source_index = index + 1
+            name = '{}{}'.format(tokens[0], index)
+        elif name.startswith('patch'):
+            index = patch_index if len(tokens) == 1 else int(tokens[1])
+            patch_index = index + 1
+            name = '{}{}'.format(tokens[0], index)
+        return name.capitalize(), patch_index, source_index, index
```

Comment:
I think this would be a bit better:

```python
name, *rest = re.split(r'(\d+)', name, 1)
sanitized_name = name.capitalize()
next_index = next_source_index if name == 'source' else next_patch_index
index = int(next(iter(rest), next_index))
if name == 'source':
    return sanitized_name, index, index + 1, patch_index
if name == 'patch':
    return sanitized_name, index, source_index, index + 1
return sanitized_name, None, source_index, patch_index
```